src/frequenz/dispatch/_actor_runner.py

```diff
@@ -0,0 +1,187 @@
+    """Helper class to manage actors based on dispatches.
+
+    Example usage:
+
+    ```python
+    import os
+    import asyncio
+    from frequenz.dispatch import Dispatcher, DispatchRunnerActor, DispatchConfigurationEvent
+    from frequenz.client.dispatch.types import ComponentSelector
+    from frequenz.client.common.microgrid.components import ComponentCategory
+
+    from frequenz.channels import Receiver, Broadcast
+    from unittest.mock import MagicMock
+
+    class MyActor(Actor):
+        def __init__(self, config_channel: Receiver[DispatchConfigurationEvent]):
+            super().__init__()
+            self._config_channel = config_channel
+            self._dry_run: bool
+            self._payload: dict[str, Any]
+
+        async def _run(self) -> None:
+            while True:
+                config = await self._config_channel.receive()
+                print("Received config:", config)
+
+                self.set_components(config.components)
+                self._dry_run = config.dry_run
+                self._payload = config.payload
+
+        def set_components(self, components: ComponentSelector) -> None:
+            match components:
+                case [int(), *_] as component_ids:
+                    print("Dispatch: Setting components to %s", components)
+                case [ComponentCategory.BATTERY, *_]:
+                    print("Dispatch: Using all battery components")
+                case _ as unsupported:
+                    print(
+                        "Dispatch: Requested an unsupported selector %r, "
+                        "but only component IDs or category BATTERY are supported.",
+                        unsupported,
+                    )
+
+    async def run():
+        url = os.getenv("DISPATCH_API_URL", "grpc://fz-0004.frequenz.io:50051")
+        key  = os.getenv("DISPATCH_API_KEY", "some-key")
+
+        microgrid_id = 1
+
+        dispatcher = Dispatcher(
+            microgrid_id=microgrid_id,
+            server_url=url,
+            key=key
+        )
+
+        # Create config channel to receive (re-)configuration events pre-start and mid-run
+        config_channel = Broadcast[DispatchConfigurationEvent](name="config_channel")
+
+        # Start actor and supporting actor, give each a config channel receiver
+        my_actor = MyActor(config_channel.new_receiver())
+        supporting_actor = MagicMock(config_channel.new_receiver())
+
+        status_receiver = dispatcher.running_status_change.new_receiver()
+
+        dispatch_handler = DispatchRunnerActor(
+            actors=frozenset([my_actor, supporting_actor]),
+            dispatch_type="EXAMPLE",
+            running_status_receiver=status_receiver,
+            configuration_sender=config_channel.new_sender(),
+        )
+
+        await asyncio.gather(dispatcher.start(), dispatch_handler.start())
+    ```
+    """
+
+    def __init__(
+        self,
+        actors: frozenset[Actor],
```

Comment:
Dunno, @daniel-zullo-frequenz seems to have required it, that part is from him :D

Reply:
It feels like it is more natural to map one actor to handling one dispatch, but I'm not strongly opposed. I would like to hear about the details on your use case thought, maybe in private to avoid spamming the PR too much?

Reply:
After talking to @daniel-zullo-frequenz we agreed that in the future his use case should use only one central actor to handle dispatches that spawns more actors as needed, but that would mean a big refactoring we can start right now, so we would leave the support for multiple actors here for now. What we can do is accept `actor: Actor | Set[Actor]`, so for all normal use cases people can pass just one actor, and document the use of multiple actors is discouraged and will be removed in the future.

Reply:
Oh, I see you removed the support for multiple actors already. I didn't meant we should remove it, but we should *figure out if we really need to ...* :(